sdk/python/lib/pulumi/runtime/stack.py

```diff
@@ -69,8 +66,11 @@
         # and each task needs to get scheduled in order to acknowledge the cancel and exit.
         await asyncio.sleep(0)
 
-        # Once we get scheduled again, all tasks have exited and we're good to go.
-        log.debug("run_in_stack completed")
+    # Once we get scheduled again, all tasks have exited and we're good to go.
+    log.debug("run_in_stack completed")
```

Comment:
Is it intentional we do this outside the `finally`?  If I follow correctly, we'll skip this now in case there is an exception thrown synchronously by the executing program?  Possibly not a big deal - but doesn't seem intentional?

Reply:
Fixed. This was indeed unintentional.